### PR TITLE
feature/geode 33

### PR DESCRIPTION
**Pull request by markito**

### CODE REVIEW

**Comment by upthewaterspout** [comment]
Hi William,

Thanks for doing all of this work! This looks like a good start for examples. I have a few comments below.

1. Did you look into using the gradle application plugin, rather than writing a custom run task?

2. We'll need to verify that the gradle wrapper jar doesn't become part of the source distribution. I guess we need to do the same thing with this gradle wrapper jar that we did at the top level of the project.

3. I wonder if this ShellUtil class should be put into some common utility project for these example tests. 

4. I  see a problem with the ShellUtil code; it's not consuming the output of the process. Especially on windows this can cause bizarre hangs because the process will block trying to write to stdout. Also it makes it really hard to debug failures if you all you do is assert that the status is 0 but don't provide the output of the process. The code should probably include the process output in the error message if the process returns a non-zero exit.

5. 'geode-examples/*/scripts/**' should not be excluded from rat.

6. How should we integrate this with the rest of the geode build? I think these examples should probably get shipped as part of the source and binary distributions, and the the examples should be built and tested as part of precheckin.

File: geode-examples/build.gradle [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        mavenCentral()
+    }
+
+    dependencies {
+        compile "org.apache.geode:geode-core:$geodeVersion"
+//        testCompile "org.apache.geode:geode-junit:$geodeVersion"
+        testCompile "junit:junit:$junitVersion"
+        testCompile "org.mockito:mockito-core:$mockitocoreVersion"
+        compile "com.jayway.awaitility:awaitility:1.7.0"
```
Comment (scmbuildguy) [inline geode-examples/build.gradle]: Add version to gradle.properties?

File: geode-examples/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+*  Have to follow code format & style from Apache Geode <sup>[1]</sup> guidelines.
+*  Should contain a `README.md` file with step-by-step instruction on how to set up and run the example. *Diagrams give you extra credit.*
+*  Donations need to be licensed through ASL 2.0 and contributors need to file an ICLA<sup>[3]</sup>.
+
+## Structure
+
+### Installation and a Tutorial for Beginners
+
+*  [How to Install](http://geode.docs.pivotal.io/docs/getting_started/installation/install_standalone.html)
+*  Set a `GEODE_HOME` environment variable to point to the `bin` directory of the installation. For those that have built from source, this will be the `geode-assembly/build/install/apache-geode` directory.
```
Comment (scmbuildguy) [inline geode-examples/README.md]: Seems like the language is unclear here: Set a `GEODE_HOME` environment variable to point to the `bin` directory

Pointing to 'bin' would not be correct as all the scripts would not work. I think wording should be changed to: 'root' directory of the installation.

File: geode-examples/replicated/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        ...
+        INFO: Done. 50 entries available on the server(s).
+
+6. Shutdown the system:
+
+        $ scripts/stopAll.sh
+
+This example is a simple demonstration on basic APIs of Geode, as well how to write tests using mocks for Geode applications.
+
+TODO: assume jUnit4
```
Comment (nabarunnag) [inline geode-examples/replicated/README.md]: TODO for an assumption rather than a future task may be incorrect  ???

**Comment by jhuynh1** [comment]
Some minor improvements/corrections:
1.) remove @author tag from Consumer.java
2.) The test checkIfScriptsAreExecutable runs the same line twice.  I think the second line was supposed to check the stop scripts and not the start scripts
3.) In one of the tests, we probably should check that Consumer.NUM_ENTRIES is > 0.  Just incase someone accidentally changes it to 0, the tests would pass but not really execute what the producer and consumer tests were expecting.
4.) preference/style : Perhaps some of the test names can be slightly more descriptive just so we know what the behavior should be.. such as changing populateWhenRegionDoesntExist to populateWhenRegionDoesNotExistShouldThrowNullPointer?

File: geode-examples/replicated/README.md (hunk 3 above)
Comment (markito) [inline geode-examples/replicated/README.md]: Oops, let's remove that line... Not needed anymore.  Thanks.

**Comment by gesterzhou** [comment]
There's no lucene in example list?

**Comment by markito** [comment]
@gesterzhou Not yet. We can definitely add to the list for 1.0.

**Comment by markito** [comment]
@jhuynh1, @gesterzhou , @nabarunnag just included the feedback items.  Thanks!

**Pull request closed without merging**